main/sources/scripture-app-builder.ts

```diff
@@ -1,18 +1,286 @@
+      ).filter((cn) => cn != null && cn !== '');
+    }
+    if (chapterNumbers.length === 0) {
+      return [];
+    }
+
+    for (const chapterNumber of chapterNumbers) {
+      const fileSelector = isPageSchema
+        ? "book[id='" + bookId + "'] > page[num='" + chapterNumber + "'] > audio > filename"
+        : "book[id='" + bookId + "'] > audio[chapter='" + chapterNumber + "'] > filename";
+
+      const timingFileSelector = isPageSchema
+        ? "book[id='" + bookId + "'] > page[num='" + chapterNumber + "'] > audio > timing-filename"
+        : "book[id='" + bookId + "'] > audio[chapter='" + chapterNumber + "'] > timing-filename";
+
+      const audioFileName = xmlDoc.querySelector(fileSelector)?.textContent;
+      const audioFileLength = xmlDoc.querySelector(fileSelector)?.getAttribute('len');
+      const timingFileName = xmlDoc.querySelector(timingFileSelector)?.textContent;
+      // cancel if anything can't be found
+      if (!audioFileName || !audioFileLength || !timingFileName || !chapterNumber) {
+        return [];
+      }
+
+      const audioFile = this.calculateAudioFile(audioFileName, directory, projectName);
+      if (!audioFile) {
+        return [];
+      }
+
+      const segments = this.makeSegments(sfmJson, +chapterNumber, timingFileName, projectName, directory);
+      if (segments.length > 0) {
+        chapters.push({
+          name: chapterNumber,
+          audio: { files: [{ filename: audioFile, length: parseInt(audioFileLength) }] },
+          segments: segments,
+        });
+      }
+    }
+
+    return chapters;
+  }
+
+  private makeSegments(
+    sfmJson: SfmJson,
+    chapterNumber: number,
+    timingFileName: string,
+    projectName: string,
+    directory: string
+  ): BKSegment[] {
+    const segments = [];
+
+    const timingFileFullName = path.join(directory, projectName, projectName + '_data', 'timings', timingFileName);
+
+    let timingFileContent;
+
+    try {
+      timingFileContent = fs.readFileSync(timingFileFullName, 'utf8');
+    } catch (error) {
+      console.error(error);
+      return [];
+    }
+
+    const timingPattern = /(\d+\.?\d*)\t(\d+\.?\d*)\t(\d+)/gs;
+
+    let timingMatches;
+    const timingOutput = [];
+    while ((timingMatches = timingPattern.exec(timingFileContent))) {
+      // console.log(timingMatches[1] + '..' + timingMatches[2] + '...' + timingMatches[3]);
+      //this is where get the start time, duration, and verse number from the timing file
+      timingOutput.push({
+        startTime: parseFloat(timingMatches[1]) * 1000,
+        endTime: parseFloat(timingMatches[2]) * 1000,
+        verse: timingMatches[3],
+      });
+    }
+
+    const chapterJson = sfmJson.chapters ? sfmJson.chapters[chapterNumber] : undefined;
+    if (!chapterJson) {
+      return [];
+    }
+    for (const verse in chapterJson) {
+      const startTime = _.find(timingOutput, { verse: verse })?.startTime;
```

Comment:
TypeScript has a simpler syntax when the variable name matches the property name. Change `{ verse: verse }` to `{ verse }`. Here and below also.

Reply:
done

Reply:
Good. There is more below in `segment.push`...

Reply:
Now it's done.